apps/extension/src/pages/formatting.tsx

```diff
@@ -0,0 +1,287 @@
+    );
+
+    if (formatting) {
+      setSelectedDateFormat(formatting.dateFormat);
+      setSelectedLanguage(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        LANGUAGE_FORMATS.find((i) => i.key === formatting.language)!,
+      );
+      setIntroSentenceType(formatting.introSentenceType);
+      setIntroSentence(formatting.introSentence);
+    } else {
+      // Reset fields
+      setSelectedDateFormat(DateFormat.MMMM_d_EEEE);
+      setIntroSentenceType(IntroSentenceType.DEFAULT);
+      resetIntroSentence(IntroSentenceType.DEFAULT);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedLanguage, formattings]);
+
+  const resetIntroSentence = (type: IntroSentenceType) => {
+    const lang = LANGUAGE_FORMATS.find(
+      ({ key }) => key === selectedLanguage.key,
+    );
+    if (type === IntroSentenceType.NONE) {
+      setIntroSentence("");
+    } else {
+      if (lang) {
+        setIntroSentence(lang.defaultIntroSentence);
+      }
+    }
+  };
+
+  const languageDropdown = (
+    <DropdownMenu.Root onOpenChange={setLanguageDropdownOpen}>
+      <DropdownMenu.Trigger>
+        <div className="h-10 bg-white color-gray-700 rounded border border-gray-200 w-64 flex py-1 px-3 justify-between items-center space-x-3 text-sm">
+          <div className="flex space-x-3 items-center">
+            <selectedLanguage.icon className="h-5 w-5" />
+            <div>{selectedLanguage.title}</div>
+          </div>
+
+          <TiArrowSortedDown className="h-6 w-6 text-gray-600" />
+        </div>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content
+        className="z-10 bg-white color-gray-700 rounded mt-2"
+        style={{ boxShadow: "0px 2px 8px rgba(0, 0, 0, 0.10)" }}
+        align="end"
+      >
+        {LANGUAGE_FORMATS.map((item) => (
+          <DropdownMenu.Item
+            key={item.key}
+            onSelect={() => {
+              if (item.key === selectedLanguage.key) return;
+              setSelectedLanguage(item);
+              setCanSave(false);
+            }}
+          >
+            <div className="h-10 bg-white color-gray-700 w-64 flex py-1 px-3 justify-between space-x-3 cursor-pointer text-sm">
+              <div className="flex space-x-3 items-center">
+                <item.icon className="h-5 w-5" />
+                <div>{item.title}</div>
+              </div>
+              {item.key === selectedLanguage.key && (
+                <HiCheckCircle className="h-4 w-4 text-primary" />
+              )}
+            </div>
+          </DropdownMenu.Item>
+        ))}
+      </DropdownMenu.Content>
+    </DropdownMenu.Root>
+  );
+
+  const introSentenceElem = (
+    <div>
+      <div className="flex border border-gray-100 rounded w-full h-9 overflow-hidden">
+        {[
+          IntroSentenceType.DEFAULT,
+          IntroSentenceType.CUSTOM,
+          IntroSentenceType.NONE,
+        ].map((type) => (
```

Comment:
```suggestion
        {Object.values(IntroSentenceType)
          .map((type) => (
```

Reply:
Good catch. I thought `Object.values` will break the order but apparently it doesn't